### PR TITLE
Emit informational msgs regarding recompilation with different GRF mode under TRITON_DEBUG

### LINKED ISSUE #2251: [Feature Improvement] Change large GRF warnings to trigger by debug flag
PR https://github.com/intel/intel-xpu-backend-for-triton/pull/1654 has been introduced using large GRF mode automatically. 

Could we make the `cout` in these lines be triggered by a debug-only flag so that normal users could safely ignore this? Those should be treated as warnings in my opinion.

https://github.com/intel/intel-xpu-backend-for-triton/blob/614efe26adeac8e28fe27c6bbfa7840bbc43ec90/third_party/intel/backend/driver.c#L188-L201

```
xpu  train AlbertForQuestionAnswering         

// We wish those won't exposed to the normal user
(I): Detected 9472 spills, recompiling the kernel using large GRF mode
(I): Kernel has now 512 spills
(I): Detected 20032 spills, recompiling the kernel using large GRF mode
(I): Kernel has now 10816 spills
(I): Detected 33600 spills, recompiling the kernel using large GRF mode
(I): Kernel has now 25408 spills
```

### PR DESCRIPTION
**Pull request by etiotto**
Fixes #2251

### CODE REVIEW

**Comment by anmyachev** [comment]
Should we hide the following warning under TRITON_DEBUG https://github.com/intel/intel-xpu-backend-for-triton/blob/750b769181deae4faf4dad3f0593c48bbfa5b214/third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp#L879 as well?

**Comment by etiotto** [comment]
> Should we hide the following warning under TRITON_DEBUG
> 
> https://github.com/intel/intel-xpu-backend-for-triton/blob/750b769181deae4faf4dad3f0593c48bbfa5b214/third_party/intel/lib/TritonGENToLLVM/TritonGENToLLVMPass.cpp#L879
> 
> as well?

This message is temporary because we have plans to switch away from GenISA intrinsics "soon". Because the message is temporary I would prefer to leave this particular message as is.

**Review by whitneywhtsang — APPROVED** [review]

File: third_party/intel/backend/driver.c [hunk 1]
```diff
@@ -192,9 +192,13 @@ static PyObject *loadBinary(PyObject *self, PyObject *args) {
     // If the register mode isn't set, and the number of spills is greater
     // than the threshold, recompile the kernel using large GRF mode.
     if (!is_GRF_mode_specified && n_spills > max_reg_spill) {
-      std::cout << "(I): Detected " << n_spills
-                << " spills, recompiling the kernel using large GRF mode"
-                << std::endl;
+      std::optional<bool> debugEnabled =
```
Comment (alexbaden) [inline third_party/intel/backend/driver.c:195]:
```suggestion
      const std::optional<bool> debugEnabled =
```

**Review by alexbaden — APPROVED** [review]

File: third_party/intel/backend/driver.c (hunk 1 above)
Comment (etiotto) [reply third_party/intel/backend/driver.c]: done

File: third_party/intel/backend/lib/libsycl-spir64-unknown-unknown.ll
Comment (whitneywhtsang) [inline third_party/intel/backend/lib/libsycl-spir64-unknown-unknown.ll:1]: @etiotto Please remove.

Comment (etiotto) [reply third_party/intel/backend/lib/libsycl-spir64-unknown-unknown.ll:1]: Done in https://github.com/intel/intel-xpu-backend-for-triton/pull/2392

**Pull request merged**